### PR TITLE
Lower transform-tree history limit from 50k to 10k

### PR DESCRIPTION
**Pull request by snosenzo**
**User-Facing Changes**

**Description**

For websocket connections with many transforms over long periods of time this limit becomes a bit more untenable, from both memory and speed perspectives. The larger the coordinate frame history the more time it takes to insert into the history, thus building up the time it takes to render a frame, which can back up the websocket. I've lowered this by 5x to keep this and memory concerns with multiple image panels in check

Expected user behavior:

Transform tree memory usage and performance improved at the cost of a shorter history for coordinate frame transforms.

FG-6239

### CODE REVIEW

**Review by achim-k — APPROVED** [review]

**Comment by defunctzombie** [comment]
> I've lowered this by 10x to keep this and memory concerns with multiple image panels in check

Is this limit "per frame id" or for all transforms?

File: packages/studio-base/src/panels/ThreeDeeRender/transforms/TransformTree.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { Transform } from "./Transform";
 import { Pose } from "./geometry";
 import { Duration, Time } from "./time";
 
-const DEFAULT_MAX_CAPACITY_PER_FRAME = 50_000;
+const DEFAULT_MAX_CAPACITY_PER_FRAME = 5_000;
```
Comment (defunctzombie) [inline packages/studio-base/src/panels/ThreeDeeRender/transforms/TransformTree.ts:10]: Once we figure out the reasoning and user-impact lets document the reason for the choice here and also link to:
https://docs.foxglove.dev/docs/visualization/panels/3d/#history

So we remember that we need to update the docs if we make this change.

**Comment by snosenzo** [comment]
@jtbandes 
> 10x seems like a big reduction. Do we recall why the limit was so high in the first place? (I don't really)

I don't quite remember why it was so high, but I have thought about lowering it before for perf concerns around transform preloading and `splice` becoming expensive as the number of transforms go up. Anyway 10X is quite a reduction, I've decided it might be better to do 5x instead and only reduce it to 10k history size. 

> Have we thought about what might need to be done to make this an overall limit instead of per-frame?

I did give it a little thought but thought it might be more work than worth it right now. I think something where an overall number is divided over the number of frames based off of the max time difference of access from the most recent transform might be a good model. So coordinate frames that have been accessed to get older timestamped tfs will receive more history budget than those that get accessed to get newer timestamped transforms. Something like that might get complicated to implement though.

File: packages/studio-base/src/panels/ThreeDeeRender/transforms/TransformTree.ts (hunk 1 above)
Comment (snosenzo) [reply packages/studio-base/src/panels/ThreeDeeRender/transforms/TransformTree.ts]: Added some considerations around changing the value in a comment

File: packages/studio-base/src/panels/ThreeDeeRender/transforms/TransformTree.ts [hunk 2]
```diff
@@ -7,7 +7,30 @@ import { Transform } from "./Transform";
...
+ *    only store 5 seconds of history for this frame.
+ *    If the object references a transform at a time older than the history, it will simply use the oldest transform for that frame
+ *    which is not guaranteed to be accurate.
+ *
+ * We generally want to keep this higher to allow for larger transform histories, but
+ * also want to be mindful to memory and performance concerns when doing so
+ *
+ * This number is mentioned in the docs. If changed docs must be updated.
+ */
+const DEFAULT_MAX_CAPACITY_PER_FRAME = 10_000;
```
Comment (defunctzombie) [inline packages/studio-base/src/panels/ThreeDeeRender/transforms/TransformTree.ts:33]: 👏

**Pull request merged**